addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -742,51 +742,77 @@ var SnippetsMenu = Widget.extend({
             return $elem.parent().width() === $elem.outerWidth(true);
         }
 
+        // Firstly, add a dropzone after the clone
+        var $clone = $('.oe_drop_clone');
+        var $drop = zone_template.clone();
+        var neighbor = $clone.prev()[0] || $clone.next()[0];
+
+        if (neighbor) {
+            var $neighbor = $(neighbor);
+            var css = window.getComputedStyle(neighbor);
+            var parentCss = window.getComputedStyle($neighbor.parent()[0]);
+            var float = css.float || css.cssFloat;
+            var parentDisplay = parentCss.display;
+            var parentFlex = parentCss.flexDirection;
+            if (float === 'left' || float === 'right' || (parentDisplay === 'flex' && parentFlex === 'row')) {
+                $drop.css('float', float);
+                if (!isFullWidth($neighbor)) {
+                    $drop.addClass('oe_vertical').css('height', Math.max($neighbor.outerHeight(), 30));
+                }
+            }
+        }
+        $clone.after($drop);
+
         if ($selectorChildren) {
             $selectorChildren.each(function () {
                 var $zone = $(this);
                 var css = window.getComputedStyle(this);
-                var parentCss = window.getComputedStyle($zone.parent()[0]);
                 var float = css.float || css.cssFloat;
-                var parentDisplay = parentCss.display;
-                var parentFlex = parentCss.flexDirection;
+                var display = css.display;
+                var flex = css.flexDirection;
```

Comment:
It was a mistake to check the parent CSS because the drop zone will be a child of $zone and not a child of the parent of $zone. It was the problem with the dropzones not vertical when necessary.

Reply:
Indeed, but that's the point: reorganize the code to reuse it, not duplicating it. Add an inline function to determine vertical or horizontal and use that inline function for the dropzone after clone, for "children" dropzone sand for "siblings" dropzones. The diff of this PR should probably be negative, not positive.

Reply:
Ok done :+1: the diff is not negative but almost ;)